sources/web/datalab/polymer/modules/GapiManager.ts

```diff
@@ -187,14 +191,43 @@ class GapiManager {
   }
 
   /**
-   * Loads the oauth2 client id from the user settings.
-   * This will change once we figure out how we want to do it.
+   * Loads the oauth2 client id. Looks first in the user settings,
+   * then in the config-local file.
    */
   private static _loadClientId(): Promise<void> {
+    return GapiManager._loadClientIdFromUserSettings()
+      .then((clientId) => {
+        if (clientId) {
+          GapiManager._clientId = clientId;
+          return;   // Make ts compiler happy
+        } else {
+          return GapiManager._loadClientIdFromConfigFile()
+            .then((clientId) => {
+              GapiManager._clientId = clientId;
+            });
+        }
+      });
+  }
+
+  /**
+   * Loads the oauth2 client id from the user settings.
+   */
+  private static _loadClientIdFromUserSettings(): Promise<string> {
     return SettingsManager.getUserSettingsAsync()
       .then((settings: common.UserSettings) => {
-        if (settings.oauth2ClientId) {
-          GapiManager._clientId = settings.oauth2ClientId;
+        return settings.oauth2ClientId;
+      });
```

Comment:
Can we add a `.catch` here that returns null or an empty string? This way the app is still usable if the user settings were messed up.